platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
@deathbaba , why do we need virtuals at all? There is always one and only one implementation of Platform on each platform. It doesn't make sense to compile PlatformWin on Android, for example. We have been used separate compilation for Platform instead of virtual inheritance and it works fine. From my point of view virtuals are not better that current approach, because you'll get a bunch of PlatformWin, PlatformLinux, PlatformPosix, PlatformUnix, ... classes instead of a bunch of source files.